test/fixtures/server.js

```diff
@@ -67,6 +70,49 @@ function createRawEchoServer () {
     return server;
 }
 
+/**
+ * Simple SSL server for tests. Send
+ *  - 'authorized' to authorized clients.
+ *  - 'unauthorized' to unauthorized clients.
+ *
+ * @param {Boolean} [requestCert=true] - If true, request a certificate from clients that connect
+ *
+ * @example
+ * var s = createSslServer();
+ * s.listen(3000, 'localhost');
+ */
+function createSslServer (requestCert) {
+    (requestCert === undefined) && (requestCert = true);
+
+    var certDataPath = path.join(__dirname, '..', 'integration-legacy', 'data'),
+        serverKeyPath = path.join(certDataPath, 'server-key.pem'),
+        serverCertPath = path.join(certDataPath, 'server-crt.pem'),
+        serverCaPath = path.join(certDataPath, 'ca-crt.pem'),
+
+        server = https.createServer({
+            key: fs.readFileSync(serverKeyPath),
+            cert: fs.readFileSync(serverCertPath),
+            ca: fs.readFileSync(serverCaPath),
+            requestCert: requestCert
+        });
+
+    server.on('request', function (req, res) {
```

Comment:
Handling all the requests in a similar manner, this won't scale well.
The aim of this server(fixtures) is to remove the boilerplate code from tests but still be able to extend its usage.

Reply:
Then how about I take the request handler as an argument too. Like below....
```javascript
function createSSLServer(opts, reqHandler){
    if(!reqHandler){
        reqHandler = function defaultReqHandler(){
            ...
        }
    }

    server.on('request', reqHandler);

}
```